File: programmers/etc/level2/피보나치_수.py
MOD = 1234567

def solution(n):
    dp = [0]*(n+1)
    dp[1] = 1
    for i in range(2, n+1):
        dp[i] = (dp[i-1]+dp[i-2])%MOD
    return dp[n]


n = 3
print(solution(n)) # 2

n = 5
print(solution(n)) # 5

File: Java/programmers/etc/level2/피보나치수.java
package CodeTest.Java.programmers.etc.level2;

public class 피보나치수 {
    public int solution(int n) {
        int MOD = 1234567;
        int[] dp = new int[n+1];
        dp[1] = 1;
        for (int i=2; i<n+1; i++) {
            dp[i] = (dp[i-1]+dp[i-2])%MOD;
        }
        return dp[n];
    }
}
